### PR TITLE
tidb-lightning: add 50TB data import best practices doc

### PR DESCRIPTION
**Pull request by niubell**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/14271
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by hfxsd** [comment]
/review 1. Evaluate the title, description, and diff of the PR. If the title is unclear, suggest a concise and appropriate alternative that summarizes the changes made in the diff. 2. Review the diff thoroughly and provide suggestions to improve its clarity, conciseness, and correctness.

File: tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    - [Issue-43291](https://github.com/pingcap/tidb/issues/43291): TiDB Lightning does not retry in cases where temporary files are not found (the "No such file or directory" error).
+
+## Prepare source files
+
+- When generating source files, within a single file, it is preferable to sort them by the primary key. If the table definition does not have a primary key, you can add an auto-increment primary key. In this case, the order of the file content does not matter.
+- When assigning source files to multiple TiDB Lightning instances, try to avoid the situation where there are overlapping primary keys or non-null unique indexes between multiple source files. If the generated files are globally sorted, they can be distributed into different TiDB Lightning instances based on ranges to achieve optimal import performance.
+- Control each file to be less than 96 MB in size during file generation.
+- If a file is exceptionally large and exceeds 256 MB, enable [strict-format](/migrate-from-csv-files-to-tidb.md#step-4-tune-the-import-performance-optional).
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md]:
```suggestion
- Control each file to be less than 96 MiB in size during file generation.
- If a file is exceptionally large and exceeds 256 MiB, enable [strict-format](/migrate-from-csv-files-to-tidb.md#step-4-tune-the-import-performance-optional).
```

File: tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+- Import only 10% of the data and multiply the actual occupied space by 10 to estimate the final space usage for that batch of data. This method is more accurate, especially when you import a large amount of data.
+
+Note that it is recommended to reserve 20% of storage space, because background tasks such as compaction and snapshot replication also consume a portion of the storage space.
+
+## Change configuration parameters
+
+- `region-concurrency`: The concurrency of TiDB Lightning main logical processing. During parallel importing, it is recommended to set it to 75% of the CPU cores to prevent resource overload and potential OOM issues.
+- `send-kv-pairs`: The number of Key-Value pairs sent by TiDB Lightning to TiKV in a single request. It is recommended to adjust this value based on the formula send-kv-pairs * row-size < 1 MB. Starting from v7.2.0, this parameter is replaced by `send-kv-size`, and no additional setting is required.
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md:95]:
```suggestion
- `send-kv-pairs`: The number of Key-Value pairs sent by TiDB Lightning to TiKV in a single request. It is recommended to adjust this value based on the formula send-kv-pairs * row-size < 1 MiB. Starting from v7.2.0, this parameter is replaced by `send-kv-size`, and no additional setting is required.
```

File: tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+title: Best Practices for Importing 50 TiB Data
+summary: Based on the experience of importing large single tables in the past, this article summarizes best practices, hoping to help you sucessfully import large data volumes and large single tables.
+---
+
+# Best Practices for Importing 50 TiB Data
+
+TiDB Lightning ([Physical Import Mode](/tidb-lightning/tidb-lightning-physical-import-mode.md)) is a comprehensive and efficient data import tool used for importing data into empty tables and initializing empty clusters, and uses files as the data source. TiDB Lightning provides two running modes: a single instance and [parallel import](/tidb-lightning/tidb-lightning-distributed-import.md). You can import source files of different sizes.
+
+- If the data scale of the source files is within 10 TiB, it is recommended to use a single instance of TiDB Lightning for the import.
+- If the data scale of the source files exceeds 10 TiB, it is recommended to use multiple instances of TiDB Lightning for [Parallel Import](/tidb-lightning/tidb-lightning-distributed-import.md).
```
Comment (Frank945946) [inline tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md:11]:
```suggestion
- If the data size of the source files exceeds 10 TiB, it is recommended to use multiple instances of TiDB Lightning for [Parallel Import](/tidb-lightning/tidb-lightning-distributed-import.md).
```

File: tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+## Resolve the "checksum mismatch" error
+
+Conflicts might occur during data validation. The error message is "checksum mismatch". To resolve this issue, take the following steps:
+
+1. In the source data, check for conflicted primary keys or unique keys, and resolve the conflicts before reimporting. In most cases, this is the most common cause.
+2. Check if the table primary key or unique key definition is reasonable. If not, modify the table definition and reimport.
+3. Enable [conflict detection](/tidb-lightning/tidb-lightning-physical-import-mode-usage.md#conflict-detection).
+
+This feature assumes that there is a small number (less than 10%) of unexpected conflicting data in the source data and requires TiDB Lightning to detect and resolve the conflicts.
```
Comment (Frank945946) [inline tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md:109]:
```suggestion
3. Enable [conflict detection](/tidb-lightning/tidb-lightning-physical-import-mode-usage.md#conflict-detection).This feature assumes that after going through the above two steps of investigation, there is a small number (less than 10%) of unexpected conflicting data in the source data, and it requires TiDB Lightning to detect and resolve these conflicts.
```

File: tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+- Check if the TiKV cluster encounters any IO bottlenecks.
+- Increase TiKV `raftstore.apply-pool-size` from the default value of `2` to `4` or `8`.
+- Reduce TiDB Lightning `region-split-concurrency` to half the number of CPU cores, with a minimum value of `1`.
+
+### Disable execution plan
+
+In the case of a large single table (for example, with over 1 billion rows and more than 50 columns), it is recommended to disable the `analyze` operation (`analyze="off"`) during the import process, and manually execute the [ANALYZE TABLE](/sql-statements//sql-statement-analyze-table.md#analyze) statement after the import is completed.
+
+For more information about the configuration of `analyze`, see [TiDB Lightning (Task)](/tidb-lightning/tidb-lightning-configuration.md#tidb-lightning-task).
```
Comment (Frank945946) [inline tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md:150]:
```suggestion
In the case of a large single table (for example, with over 1 billion rows and more than 50 columns), it is recommended to disable the `analyze` operation (`analyze="off"`) during the import process（For more information about the configuration of `analyze`, see [TiDB Lightning (Task)](/tidb-lightning/tidb-lightning-configuration.md#tidb-lightning-task).）, and manually execute the [ANALYZE TABLE](/sql-statements//sql-statement-analyze-table.md#analyze) statement after the import is completed.

```

Comment (hfxsd) [reply tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md:148]: Keep it as is after discussion.

File: tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
+---
+title: Best Practices for Importing 50 TiB Data
+summary: Based on the experience of importing large single tables in the past, this article summarizes best practices, hoping to help you sucessfully import large data volumes and large single tables.
```
Comment (lilin90) [inline tidb-lightning/tidb-lightning-50TB-data-import-best-practices.md:3]:
```suggestion
summary: Based on the experience of importing large single tables in the past, this article summarizes best practices, hoping to help you successfully import large data volumes and large single tables.
```

**Review by hfxsd — APPROVED** [review]

File: tidb-lightning/data-import-best-practices.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+- Check whether the TiKV cluster encounters any I/O bottlenecks.
+- Increase TiKV `raftstore.apply-pool-size` from the default value of `2` to `4` or `8`.
+- Reduce TiDB Lightning `region-split-concurrency` to half the number of CPU cores, with a minimum value of `1`.
+
+### Disable the execution plan
+
+In the case of a large single table (for example, with over 1 billion rows and more than 50 columns), it is recommended to disable the `analyze` operation (`analyze="off"`) during the import process, and manually execute the [`ANALYZE TABLE`](/sql-statements//sql-statement-analyze-table.md) statement after the import is completed.
+
+For more information about the configuration of `analyze`, see [TiDB Lightning (Task)](/tidb-lightning/tidb-lightning-configuration.md#tidb-lightning-task).
```
Comment (lilin90) [inline tidb-lightning/data-import-best-practices.md:151]:
```suggestion
For more information about the configuration of `analyze`, see [TiDB Lightning task configuration](/tidb-lightning/tidb-lightning-configuration.md#tidb-lightning-task).
```

**Review by lilin90 — COMMENTED** [review]
/lgtm

**Pull request merged**